pkg/http/sse.go

```diff
@@ -2,22 +2,28 @@ package http
 
 import (
 	"context"
+	"net/http"
+
 	"github.com/go-chi/render"
 	"github.com/pkg/errors"
-	"net/http"
 
 	"github.com/ThreeDotsLabs/watermill"
 	"github.com/ThreeDotsLabs/watermill/message"
 	"github.com/ThreeDotsLabs/watermill/pubsub/gochannel"
 )
 
 type StreamAdapter interface {
-	// GetResponse returns the response to be sent back to client.
-	// Any errors that occur should be handled and written to `w`, returning false as `ok`.
-	GetResponse(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
-	// Validate validates if the incoming message should be handled by this handler.
-	// Typically this involves checking some kind of model ID.
-	Validate(r *http.Request, msg *message.Message) (ok bool)
+	// InitialStreamResponse returns the first stream response to be sent back to client.
+	// Any errors that occur should be handled and written to `w`.
+	// Returning `ok` equal false ends processing the HTTP request.
+	InitialStreamResponse(w http.ResponseWriter, r *http.Request) (response interface{}, ok bool)
+	// NextEvent returns the next stream response to be sent back to the client.
```

Comment:
maybe NextStreamResponse instead of NextEvent will be better name? to make it a bit more consistent